actors/test/terminate_sectors_scenario_test.go

```diff
@@ -0,0 +1,246 @@
+
+	terminateParams := miner.TerminateSectorsParams{
+		Terminations: []miner.TerminationDeclaration{{
+			Deadline:  dlInfo.Index,
+			Partition: pIdx,
+			Sectors:   bitfield.NewFromSet([]uint64{uint64(sectorNumber)}),
+		}},
+	}
+
+	_, code = v.ApplyMessage(worker, minerAddrs.RobustAddress, big.Zero(), builtin.MethodsMiner.TerminateSectors, &terminateParams)
+	require.Equal(t, exitcode.Ok, code)
+
+	noSubinvocations := []vm.ExpectInvocation{}
+	vm.ExpectInvocation{
+		To:     minerAddrs.IDAddress,
+		Method: builtin.MethodsMiner.TerminateSectors,
+		SubInvocations: []vm.ExpectInvocation{
+			{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward, SubInvocations: noSubinvocations},
+			{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower, SubInvocations: noSubinvocations},
+			{To: builtin.BurntFundsActorAddr, Method: builtin.MethodSend, SubInvocations: noSubinvocations},
+			{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.UpdatePledgeTotal, SubInvocations: noSubinvocations},
+			{To: builtin.StorageMarketActorAddr, Method: builtin.MethodsMarket.OnMinerSectorsTerminate, SubInvocations: noSubinvocations},
+			{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.UpdateClaimedPower, SubInvocations: noSubinvocations},
+		},
+	}.Matches(t, v.LastInvocation())
+
+	// expect power, market and miner to be in base state
+	minerBalances := vm.GetMinerBalances(t, v, minerAddrs.IDAddress)
+	assert.Equal(t, big.Zero(), minerBalances.InitialPledge)
+	assert.Equal(t, big.Zero(), minerBalances.PreCommitDeposit)
+
+	// expect network stats to reflect power has been removed from sector
+	stats := vm.GetNetworkStats(t, v)
+	assert.Equal(t, int64(0), stats.MinerAboveMinPowerCount)
+	assert.Equal(t, big.Zero(), stats.TotalRawBytePower)
+	assert.Equal(t, big.Zero(), stats.TotalQualityAdjPower)
+	assert.Equal(t, big.Zero(), stats.TotalBytesCommitted)
+	assert.Equal(t, big.Zero(), stats.TotalQABytesCommitted)
+	assert.Equal(t, big.Zero(), stats.TotalPledgeCollateral)
+
+	// market cron slashes deals because sector has been terminated
+	for _, id := range dealIDs {
+		state, found := vm.GetDealState(t, v, id)
+		require.True(t, found)
+		// non-zero
+		assert.Greater(t, uint64(state.LastUpdatedEpoch), uint64(0))
+		// deal has not been slashed
+		assert.Equal(t, v.GetEpoch(), state.SlashEpoch)
+
+	}
+
+	// advance and run cron to complete processing of termination
+	v, err = v.WithEpoch(v.GetEpoch() + 1000)
+	require.NoError(t, err)
+	_, code = v.ApplyMessage(builtin.SystemActorAddr, builtin.CronActorAddr, big.Zero(), builtin.MethodsCron.EpochTick, nil)
+	require.Equal(t, exitcode.Ok, code)
+
+	// Verified client should be able to withdraw all all deal collateral.
+	// Client added 3 FIL balance and had 2 deals with 1 FIL collateral apiece.
+	// Should only be able to withdraw the full 2 FIL if deals have been slashed and balance was unlocked.
+	withdrawal := big.Mul(big.NewInt(2), vm.FIL)
```

Comment:
What about the 3 FIL of own balance. It should be able to withdraw 5, I think.

Reply:
Verify that the miner has lost escrow balance?

Reply:
> What about the 3 FIL of own balance. It should be able to withdraw 5, I think.

The verified client only added 3 FIL collateral to the market actor, so the market actor can't send more than that to verified client's account (which is what we're testing). Requesting 5 FIL actually sends 2999999994151043072 AttoFIL which is all the collateral added minus payment for the time the deals were stored before they were slashed.